data-raw/ictr/ictr_main.Rmd

```diff
@@ -0,0 +1,293 @@
+Product Carbon Transition Risk Indicator, but it focuses on the input products
+and not the product of the company. Input products are, for example, resources,
+packaging materials, energy and enabling services (such as tractor use on farm)
+to produce the product.
+
+After identifying each carbon footprint for one input product, the input
+products will be ranked according to their footprint. Input products in the
+highest percentile (≥70%) will be classified as high transition risk input
+products. Input products in the medium percentile (between ≥30% and <70%) will
+be classified as medium transition risk input products. Products in the lowest
+percentile (<30%) will be classified as low transition risk input products.
+
+After assessing the input products transition risk based on its carbon footprint
+for each product, they will be aggregated at company-level. We derive what
+percentage of the input products are high, medium and low transition risk.
+
+The goal of the transition risk MVP is to create a first draft of how the
+transition risk indicator for inputs would be build in code in order to convert
+it into code production easily in the future.
+
+The transition risk indicator consists of 3 main steps:
+
+- Step 1: Identifying the input products for each product
+- Step 2: Calculating the relative carbon footprint per input product
+- Step 3: Aggregating on the company-level
+
+# Step 1: Identifying
+
+Identifying the input products for each product
+
+## 1.1 Load company data and inputs data
+
+Sample data set includes inputs and co2 footprints for each product from
+Ecoinvent and sectors from Europages. NOTE: the following columns are a complete
+random selection and do not reflect the true information:
+
+    - co2 footprints (not allowed to share licensed data right now)
+    - sectors (as the matching with ecoinvent is not done yet, we do not have one sector per product yet)
+
+```{r}
+inputs <- read.csv(here("data_prep","02_input_data", "data_ecoinvent_co2.csv"))
+companies <- read.csv(here("data_prep","02_input_data", "tilt_companies.csv"))
+```
+
+# Step 2: Calculate ranks per input product and assign scores
+
+## 2.1 Calculate the rank of each input product
+
+This is done based on the input products' carbon footprints compared to 4
+different benchmarks (all products, products with same unit, products with same
+sector, products with same unit and sector)
+
+```{r}
+# rank in comparison to all input products
+ecoinvent_input <- inputs %>%
+  mutate(perc_all = rank(input_co2) / length(input_co2))
+
+# rank in comparison to all input products with same unit
+ecoinvent_input <- ecoinvent_input |>
+  group_by(unit) |>
+  mutate(perc_unit = rank(input_co2) / length(input_co2)) %>%
+  ungroup()
+
+# rank in comparison to all input products with same input sector
+ecoinvent_input <- ecoinvent_input |>
+  group_by(input_sector) |>
+  mutate(perc_sec = rank(input_co2) / length(input_co2)) %>%
+  ungroup()
```

Comment:
In beneral it's best to [avoid overwriting intermediate results.](https://gist.github.com/maurolepore/17b468dd8c3a603c668220563754ad7f#avoid-overwriting-intermediate-results). Here you may simply connect each piece with the pipe. If the intermediate results are meaningful, then express that meaning in the name of each intermediate result.